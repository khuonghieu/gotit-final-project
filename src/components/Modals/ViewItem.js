import React from 'react';
import { connect } from 'react-redux';
import { Modal } from '@gotitinc/design-system';

function mapStateToProps(state) {
  return {
    item: state.item,
  };
}

export function ViewItem({ onClose, item }) {
  return (
    <Modal size="small" relative centered onHide={onClose}>
      <Modal.Header closeButton onClick={onClose}>
        <Modal.Title>Item Details</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <div className="u-textCenter">
          <img src="holder.js/100px90?text=Image" className="u-maxWidthFull u-marginBottomExtraSmall" alt="" />
        </div>
        <div>
          Item name:
          {' '}
          {item.name}
        </div>
        <div>
          Item description:
          {' '}
          {item.description}
        </div>
        <div>
          Item price:
          {' '}
          {item.price}
        </div>
      </Modal.Body>
    </Modal>
  );
}

export default connect(mapStateToProps, null)(ViewItem);
