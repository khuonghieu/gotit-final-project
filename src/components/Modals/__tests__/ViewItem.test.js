import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Form, Button } from '@gotitinc/design-system';
import { ViewItem } from '../ViewItem';

configure({ adapter: new Adapter() });

describe('components/Modals/ViewItem.js', () => {
  let wrapper;
  let props;
  let input;
  let button;

  beforeEach(() => {
    props = {
      onClose: jest.fn(),
      item: {
        name: 'testname',
        description: 'testdescription',
        id: 0,
        price: 0,
      },
    };
  });

  const update = () => {
    input = wrapper.find(Form.Input);
    button = wrapper.find(Button);
  };

  const setup = () => {
    wrapper = shallow(
      <ViewItem {...props} />,
    );
    update();
  };

  it('should render correctly', () => {
    setup();
    expect(wrapper).toMatchSnapshot();
  });

  // it('should close when click close button', () => {
  //   setup();
  //   button.simulate('click');
  //   expect(props.onClose).toHaveBeenCalled();
  // });
});
