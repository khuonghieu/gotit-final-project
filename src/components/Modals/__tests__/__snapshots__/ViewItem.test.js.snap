// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Modals/ViewItem.js should render correctly 1`] = `ShallowWrapper {}`;
